File: modules/agent/src/mocks/services/orderService.ts
import * as orderRepo from '../repositories/orderRepo.js';
import * as customerRepo from '../repositories/customerRepo.js';

interface OrderItem {
  sku: string;
  qty: number;
  unit_price: number;
  location: string;
}

export const getOrders = (status?: string, customerId?: string) => {
  let orders = orderRepo.getAllOrders();

  if (status) {
    orders = orders.filter((o) => o.status === status);
  }
  if (customerId) {
    orders = orders.filter((o) => o.customer_id === customerId);
  }

  return orders;
};

export const getOrderDetail = (orderId: string) => {
  const order = orderRepo.findOrderById(orderId);
  if (!order) return null;

  const customer = customerRepo.findCustomerById(order.customer_id);

  return {
    ...order,
    customer_name: customer?.name,
    customer_email: customer?.email,
    shipping_address: customer?.shipping_address,
  };
};

export const createOrder = (customerId: string, items: OrderItem[]) => {
  const customer = customerRepo.findCustomerById(customerId);
  if (!customer) {
    return { error: `Customer ${customerId} not found` };
  }

  const orderId = `ORD-${10000 + orderRepo.getOrderCount() + 1}`;
  const total = items.reduce((sum, item) => sum + item.qty * item.unit_price, 0);

  const newOrder = {
    id: orderId,
    customer_id: customerId,
    status: 'pending',
    created_at: new Date().toISOString(),
    shipped_at: null,
    items,
    total,
  };

  return orderRepo.appendOrder(newOrder);
};

export const updateOrderStatus = (orderId: string, newStatus: string) => {
  const validStatuses = orderRepo.getValidStatuses();
  if (!validStatuses.includes(newStatus)) {
    return { error: `Invalid status. Must be one of: ${validStatuses.join(', ')}` };
  }

  const order = orderRepo.findOrderById(orderId);
  if (!order) {
    return { error: `Order ${orderId} not found` };
  }

  order.status = newStatus;
  if (newStatus === 'shipped') {
    order.shipped_at = new Date().toISOString();
  }

  return order;
};

export const getOrderSummary = () => {
  const orders = orderRepo.getAllOrders();
  const validStatuses = orderRepo.getValidStatuses();

  const summary: Record<string, number> = {};
  for (const status of validStatuses) {
    summary[status] = 0;
  }

  let totalValue = 0;

  for (const order of orders) {
    summary[order.status] = (summary[order.status] ?? 0) + 1;
    totalValue += order.total;
  }

  return {
    by_status: summary,
    total_orders: orders.length,
    total_value: totalValue,
  };
};

export const getCustomers = () => {
  return customerRepo.getAllCustomers();
};

File: modules/agent/src/agent/mocks/services/order_service.py
from datetime import datetime

from agent.mocks.repositories import order_repo, customer_repo


def get_orders(status: str | None = None, customer_id: str | None = None) -> list[dict]:
    orders = order_repo.get_all_orders()

    if status:
        orders = [o for o in orders if o["status"] == status]
    if customer_id:
        orders = [o for o in orders if o["customer_id"] == customer_id]

    return orders


def get_order_detail(order_id: str) -> dict | None:
    order = order_repo.find_order_by_id(order_id)
    if not order:
        return None

    customer = customer_repo.find_customer_by_id(order["customer_id"]) or {}

    return {
        **order,
        "customer_name": customer.get("name"),
        "customer_email": customer.get("email"),
        "shipping_address": customer.get("shipping_address"),
    }


def create_order(customer_id: str, items: list[dict]) -> dict:
    customer = customer_repo.find_customer_by_id(customer_id)
    if not customer:
        return {"error": f"Customer {customer_id} not found"}

    order_id = f"ORD-{10000 + order_repo.get_order_count() + 1}"
    total = sum(item["qty"] * item["unit_price"] for item in items)

    new_order = {
        "id": order_id,
        "customer_id": customer_id,
        "status": "pending",
        "created_at": datetime.now().isoformat() + "Z",
        "shipped_at": None,
        "items": items,
        "total": total,
    }

    return order_repo.append_order(new_order)


def update_order_status(order_id: str, new_status: str) -> dict:
    valid_statuses = order_repo.get_valid_statuses()
    if new_status not in valid_statuses:
        return {"error": f"Invalid status. Must be one of: {valid_statuses}"}

    order = order_repo.find_order_by_id(order_id)
    if not order:
        return {"error": f"Order {order_id} not found"}

    order["status"] = new_status
    if new_status == "shipped":
        order["shipped_at"] = datetime.now().isoformat() + "Z"

    return order


def get_order_summary() -> dict:
    orders = order_repo.get_all_orders()
    valid_statuses = order_repo.get_valid_statuses()

    summary = {status: 0 for status in valid_statuses}
    total_value = 0

    for order in orders:
        summary[order["status"]] += 1
        total_value += order["total"]

    return {
        "by_status": summary,
        "total_orders": len(orders),
        "total_value": total_value,
    }


def get_customers() -> list[dict]:
    return customer_repo.get_all_customers()
